homeassistant/components/shelly/entity.py

```diff
@@ -472,6 +536,58 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
         return self.description.extra_state_attributes(self.wrapper.device.status)
 
 
+class ShellyRpcAttributeEntity(ShellyRpcEntity, entity.Entity):
+    """Helper class to represent a rpc attribute."""
+
+    def __init__(
+        self,
+        wrapper: RpcDeviceWrapper,
+        key: str,
+        attribute: str,
+        description: RpcAttributeDescription,
+    ) -> None:
+        """Initialize sensor."""
+        super().__init__(wrapper, key)
+        self.attribute = attribute
+        self.description = description
+
+        self._attr_unique_id = f"{super().unique_id}-{attribute}"
+        self._attr_name = get_rpc_entity_name(wrapper.device, key, description.name)
+        self._attr_entity_registry_enabled_default = description.default_enabled
+        self._attr_device_class = description.device_class
+        self._attr_icon = description.icon
+        self._last_value = None
+
+    @property
+    def attribute_value(self) -> StateType:
+        """Value of sensor."""
+        if callable(self.description.value):
+            self._last_value = self.description.value(
+                self.wrapper.device.status[self.key], self._last_value
+            )
+        return self._last_value
+
+    @property
+    def available(self) -> bool:
+        """Available."""
+        available = super().available
+
+        if not available or not self.description.available:
+            return available
+
+        return self.description.available(self.wrapper.device.status[self.key])
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
```

Comment:
Can be switched to `self._attr_extra_state_attributes` ?

Reply:
Same as previous answer